stylix/palette.nix

```
@@ -151,6 +152,13 @@ in
     lib.stylix.colors = (base16.mkSchemeAttrs cfg.base16Scheme).override cfg.override;
     lib.stylix.scheme = base16.mkSchemeAttrs cfg.base16Scheme;
 
+    assertions = [
+      {
+        assertion = cfg.image != null || cfg.base16Scheme != null;
```

I find this variant easier to understand:

```suggestion
        assertion = cfg.image == null && cfg.base16Scheme == null;
```

---

aren't assertions supposed to be true? we don't want both to be nul

---

> aren't assertions supposed to be true? we don't want both to be null

That's the point. Both predicates should evaluate to the same according to [De Morgan's laws](https://en.wikipedia.org/wiki/De_Morgan's_laws). However, my proposal might be easier to understand.

---

wouldn't your version only be true when both are null? i'll make some tables table to wrap my head around this

**x != null || y != null**
|  | null | set |
|--------|--------|--------|
| null | false | true |
| set | true | true |

**x == null && y == null**
|  | null | set |
|--------|--------|--------|
| null | true | false |
| set | false | false |

so according to these tables, you've negated my predicate, and I think De Morgan's laws agree:
> The rules can be expressed in English as:
>
> - The negation of "A and B" is the same as "not A or not B".
> - The negation of "A or B" is the same as "not A and not B".